bspi.py

```diff
@@ -5,24 +5,40 @@
 import re
 import subprocess
 import sys
+import argparse
+import configparser
+import xdg
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+default_config_loc = f'{script_dir}/bspi_iconmap.ini'
+
+parser = argparse.ArgumentParser(description='bspi - rename bspwm desktops with icons based on the applications \
+    currently running under that desktop')
+parser.add_argument('-c', '--config', default=f'{default_config_loc}', help='absolute or relative path to the bspi \
+    configuration file')
+args = parser.parse_args()
 
-ICONS = {
-        "dolphin": "",
-        "chromium": "",
-        "firefox": "",
-        "signal": "",
-        "slack": "",
-        "spotify": "",
-        os.environ.get('TERMINAL'): "",
-        None: " "
-    }
 
 class Icon:
-    def __init__(self, client_class):
+    def __init__(self, client_class=None):
         self.client_class = client_class
+        self.ICONS = self._config()
 
     def __str__(self):
-        return ICONS.get(self.client_class.lower(), ICONS[None])
+        return self.ICONS.get(self.client_class.lower(), self.ICONS.get('None'))
+
+    def _config(self):
+        config = configparser.ConfigParser()
+        if os.path.isfile(args.config):
+            cfg_file_path = args.config
+
+        else:
+            cfg_file_path = f'{xdg.XDG_CONFIG_HOME}/bspwm/bspi_iconmap.ini'
```

Comment:
Help suggests there is a default config path, that points to the stock configuration. 
Then `XDG_CONFIG_HOME` fallback works only if the explicitly is provided *and* does not exist - that's a bit confusing.

I think we can give up on xdg dependency altogether, plain config overwrite via command line arg is IMHO enough. What do you think?

Reply:
Thanks for the feedback, it was precisely something like that I felt I had done, but didn't actually have time to untangle the logic when I was writing this. 

Command line arg is definitely enough for this, and removes the external dependency on pip and the pip-package xdg. Much better :+1: